### PR TITLE
Added support for SubResource Integrity and Cross Origin attribute

### PR DESCRIPTION
**Pull request by cwchong**
Added support for SubResource Integrity and Cross Origin attribute in Requirements::css.

Requirements::css($file, $media = null, $options = []);

$options = [
    'integrity' => 'the_generated_resource_hash',
    'crossorigin' => 'cross_origin_policy_for_the_resource'
];

### CODE REVIEW

File: src/View/Requirements_Backend.php [hunk 1]
```diff
@@ -631,13 +631,21 @@ public function javascriptTemplate($file, $vars, $uniquenessID = null)
...
+     * - 'integrity' : SubResource Integrity hash
+     * - 'crossorigin' : Cross-origin policy for the resource
      */
-    public function css($file, $media = null)
+    public function css($file, $media = null, $options = [])
     {
         $file = ModuleResourceLoader::singleton()->resolvePath($file);
 
+        $integrity = isset($options['integrity']) ? $options['integrity'] : null;
+        $crossorigin = isset($options['crossorigin']) ? $options['crossorigin'] : null;
```
Comment (robbieaverill) [inline src/View/Requirements_Backend.php]: We have PHP 7.1 as a minimum in this branch now, so you could do `$crossorigin = $options['crossorigin'] ?? null;` to shorten it a little

**Comment by cwchong** [comment]
> Looks like a reasonable change. Do you want to do this for the `javascript` API as well? Also a couple of basic unit tests would be nice if you could stretch that far =)

Yes I can try to do that but I would need some guidance on getting the base tests to even work, am probably doing it wrong:

Installed via 
composer create-project --keep-vcs --dev silverstripe/installer ./ 4.x-dev --prefer-source

Ran test without any changes via 
.\vendor\bin\phpunit .\vendor\silverstripe\framework\tests\php\View\RequirementsTest.php

Then got a bunch of test failures

**Comment by robbieaverill** [comment]
That's odd. Feel free to ping me on the community Slack channel to discuss further if you'd like to.

**Comment by Juanitou** [comment]
Hi! I see the PR is closed, but I'm curious about the SRI feature in the ```javascript()``` function. Did you manage to work on it?

**Comment by Juanitou** [comment]
Sorry for the noise, I didn't see #9139!

**Pull request closed without merging**